Wikipedia/Code/UserHistoryFunnel.swift

```diff
@@ -89,3 +119,48 @@
         // DDLogDebug("Attempted to log starting User History snapshot")
     }
 }
+
+private extension WMFContentGroupKind {
+    var offLanguageCodes: Set<String> {
+        let preferredLangCodes = MWKLanguageLinkController.sharedInstance().preferredLanguages.map{$0.languageCode}
+        return Set(preferredLangCodes).subtracting(languageCodes)
+    }
+    
+    // codes define by: https://meta.wikimedia.org/wiki/Schema:MobileWikiAppiOSUserHistory
+    var userHistorySchemaCode: String? {
```

Comment:
per our chat i made this optional

i also renamed this and the next var since they're specific to this funnel's schema (which is also why i moved it into this file and made the entire extension private)